scripts/run_presubmit_checks_test.py

```diff
@@ -0,0 +1,152 @@
+
+from scripts import run_backend_tests
+from scripts import run_frontend_tests
+from scripts import run_presubmit_checks
+from scripts.linters import pre_commit_linter
+
+
+class RunPresubmitChecksTests(test_utils.GenericTestBase):
+    """Unit tests for scripts/run_presubmit_checks.py."""
+
+    def setUp(self) -> None:
+        super(RunPresubmitChecksTests, self).setUp()
+        self.print_arr: list[str] = []
+        def mock_print(msg: str) -> None:
+            self.print_arr.append(msg)
+        self.print_swap = self.swap(builtins, 'print', mock_print)
+
+        current_dir = os.path.abspath(os.getcwd())
+        self.changed_frontend_file = os.path.join(
+            current_dir, 'core', 'templates', 'tesFile.ts')
+        self.current_branch = 'test-branch'
+        self.cmd_to_check_current_branch = [
+            'git', 'rev-parse', '--abbrev-ref', 'HEAD']
+        self.cmd_to_match_current_branch_with_remote = [
+        'git', 'ls-remote', '--heads', 'origin', self.current_branch, '|',
+        'wc', '-l']
+
+        self.scripts_called = {
+            'run_frontend_tests': False,
+            'run_backend_tests': False,
+            'pre_commit_linter': False
+        }
+        def mock_frontend_tests(args: list[str]) -> None:  # pylint: disable=unused-argument
+            self.scripts_called['run_frontend_tests'] = True
+        def mock_backend_tests(args: list[str]) -> None:  # pylint: disable=unused-argument
+            self.scripts_called['run_backend_tests'] = True
+        def mock_pre_commit_linter(args: list[str]) -> None:  # pylint: disable=unused-argument
+            self.scripts_called['pre_commit_linter'] = True
+
+        self.swap_frontend_tests = self.swap(
+            run_frontend_tests, 'main', mock_frontend_tests)
+        self.swap_backend_tests = self.swap(
+            run_backend_tests, 'main', mock_backend_tests)
+        self.swap_pre_commit_linter = self.swap(
+            pre_commit_linter, 'main', mock_pre_commit_linter)
+
+    def test_run_presubmit_checks_when_branch_is_specified(self) -> None:
+        specified_branch = 'develop'
+        cmd_to_get_all_changed_file = [
+        'git', 'diff', '--cached', '--name-only', '--diff-filter=ACM',
+        specified_branch]
+        def mock_check_output(cmd: list[str]) -> str:
+            if cmd == self.cmd_to_check_current_branch:
+                return self.current_branch
+            elif cmd == self.cmd_to_match_current_branch_with_remote:
+                return '1'
+            elif cmd == cmd_to_get_all_changed_file:
+                return self.changed_frontend_file
+            else:
+                return ''
+
+        swap_check_output = self.swap(
+            subprocess, 'check_output', mock_check_output)
+        with self.print_swap, swap_check_output, self.swap_pre_commit_linter:
+            with self.swap_backend_tests, self.swap_frontend_tests:
+                run_presubmit_checks.main(args=['--b', specified_branch])
```

Comment:
```suggestion
                run_presubmit_checks.main(args=['-b', specified_branch])
```

We normally only use one hyphen when using the single-letter abbreviations

Reply:
Fixed